runtime/src/iree/hal/drivers/hip/hip_device.c

```diff
@@ -1335,15 +1340,27 @@ static iree_status_t iree_hal_hip_device_complete_buffer_operation(
 
   if (data->buffer &&
       data->type == IREE_HAL_HIP_DEVICE_SEMAPHORE_OPERATION_ASYNC_DEALLOC) {
-    int device_ordinal =
-        iree_math_count_trailing_zeros_u64(data->base.queue_affinity);
-    if (data->base.device->supports_memory_pools) {
-      status = iree_status_join(
-          status,
-          iree_hal_hip_memory_pools_deallocate(
-              &data->base.device->devices[device_ordinal].memory_pools,
-              data->base.device->devices[device_ordinal].hip_dispatch_stream,
-              data->buffer));
+    iree_hal_allocator_t* device_allocator =
```

Comment:
I think the logic here is inconsistent with allocation.
If we support memory pools that we *always* use the memory pools, regardless of the allocator that as attached.

If we are not using memory pools then what we do depends on if we have a wrapped buffer or not.